File: app/src/main/java/com/lab601/loopicandroid/module/ConfigManager.java
package com.lab601.loopicandroid.module;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class ConfigManager {

    private static ConfigManager mInstance;
    private boolean landscape = true;   //横屏
    private boolean sound = false;  //静音
    public static HashMap<Integer, String> preloadMap = new HashMap<>();

    private int startIndex = 0;
    private List<String> text = null;
    private String url = "";

    private ConfigManager() {
        text = new ArrayList<>();
    }

    public static ConfigManager getInstance() {
        if (mInstance == null) {
            mInstance = new ConfigManager();
        }
        return mInstance;
    }

    public boolean isLandscape() {
        return landscape;
    }

    public void setLandscape(boolean landscape) {
        this.landscape = landscape;
    }

    public boolean isSound() {
        return sound;
    }

    public void setSound(boolean sound) {
        this.sound = sound;
    }

    public int getStartIndex() {
        return startIndex;
    }

    public ConfigManager setStartIndex(int startIndex) {
        this.startIndex = startIndex;
        return this;
    }

    public String getUrl() {
        return url;
    }

    public ConfigManager setUrl(String url) {
        this.url = url;
        return this;
    }

    public List<String> getText() {
        return text;
    }

    public ConfigManager setText(List<String> text) {
        this.text = text;
        return this;
    }
}

File: app/src/main/java/com/lab601/loopicandroid/module/ConfigManager.kt
package com.lab601.loopicandroid.module

import com.lab601.loopicandroid.api.LooService
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory
import java.util.*


object ConfigManager{
    var isLandscape = true //横屏
    var isSound = false //静音
    var startStory = 0
    var startScene = 0
    var text: List<String>? = null
    var currSceneList:List<String>? = null  //当前选中的Scene列表

    var storyList:List<String>? = null
    var allSceneList:List<List<String>>? = null

    var url = ""
    lateinit var service: LooService

    var preloadMap = HashMap<Int, String>()

    init {
        text = ArrayList()
    }

    fun initNetwork(ip:String){
        url = ip
        val retrofit = Retrofit.Builder()
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .baseUrl("http://${ip}:8080/")
                .build()
        service = retrofit.create(LooService::class.java)
    }
}